ui/apps/everest/src/pages/databases/DbTableActionsMenu/DbTableActionsMenu.tsx

```diff
@@ -0,0 +1,245 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useState } from 'react';
+import { Box, IconButton, Menu, MenuItem } from '@mui/material';
+import MoreHorizOutlinedIcon from '@mui/icons-material/MoreHorizOutlined';
+import { MRT_Row } from 'material-react-table';
+import { DbClusterTableElement } from '../dbClusterView.types';
+import { Link } from 'react-router-dom';
+import {
+  BorderColor,
+  DeleteOutline,
+  PauseCircleOutline,
+} from '@mui/icons-material';
+import { Messages } from '../dbClusterView.messages';
+import RestartAltIcon from '@mui/icons-material/RestartAlt';
+import AddIcon from '@mui/icons-material/Add';
+import KeyboardReturnIcon from '@mui/icons-material/KeyboardReturn';
+import { DbCluster, DbClusterStatus } from 'shared-types/dbCluster.types';
+import { useDbActions } from 'hooks/api/db-cluster/useDbActions';
+import { useGetPermissions } from 'utils/useGetPermissions';
+
+const DbActionButtons = (
+  row: MRT_Row<DbClusterTableElement>,
+  closeMenu: (event: React.MouseEvent) => void,
+  setIsNewClusterMode: React.Dispatch<React.SetStateAction<boolean>>,
+  handleDbRestart: (dbCluster: DbCluster) => void,
+  handleDbSuspendOrResumed: (dbCluster: DbCluster) => void,
+  handleDeleteDbCluster: (dbCluster: DbCluster) => void,
+  isPaused: (dbCluster: DbCluster) => boolean | undefined,
+  handleRestoreDbCluster: (dbCluster: DbCluster) => void,
+  canCreate: boolean
+) => {
+  const { canUpdate, canDelete } = useGetPermissions({
+    resource: 'database-clusters',
+    specificResource: row.original.databaseName,
+    namespace: row.original.namespace,
+  });
+
+  return [
+    <MenuItem
+      disabled={row.original.status === DbClusterStatus.restoring}
+      key={0}
+      component={Link}
+      onClick={(event) => {
+        closeMenu(event);
+      }}
+      to="/databases/edit"
+      state={{
+        selectedDbCluster: row.original.databaseName!,
+        namespace: row.original.namespace,
+      }}
+      sx={{
+        m: 0,
+        display: canUpdate ? 'flex' : 'none',
+        gap: 1,
+        alignItems: 'center',
+        px: 2,
+        py: '10px',
+      }}
+    >
+      <BorderColor fontSize="small" /> {Messages.menuItems.edit}
+    </MenuItem>,
+    <MenuItem
+      disabled={row.original.status === DbClusterStatus.restoring}
+      key={2}
+      onClick={(event) => {
+        handleDbRestart(row.original.raw);
+        closeMenu(event);
+      }}
+      sx={{
+        m: 0,
+        display: canUpdate ? 'flex' : 'none',
```

Comment:
Let's not use display to control what a user sees or not. Imagine a dev just opens the dev tools and they can tell what what actions are available, even though it's not for them.

This should be done instead:

```
[
...(canUpdate ? [<MenuItem />] : [])
]
```